src/configuration.ts

```diff
@@ -5,18 +5,20 @@ interface LanguageConfig {
     readonly markupCompletionsEnabled?: boolean;
     readonly showOnColon?: boolean;
     readonly emojiDecoratorsEnabled?: boolean;
+    readonly enableForAllLanguages?: boolean;
 }
 
 export class Configuration implements LanguageConfig {
 
-    public get languages(): Iterable<string> {
-        return this.languageConfigurations.keys();
+    public async getLanguages(): Promise<Iterable<string>> {
```

Comment:
Instead of making this async, can you resolve `languages` once in the `activate` function?

Reply:
It's tit for tat, since now we need to make `updateConfiguration` async and calling it in the constructor is redundant due to async nature of method. Let me know what you think.